.eslintrc.js

```diff
@@ -4,12 +4,31 @@ module.exports = {
     'mocha': true,
     'es6': true
   },
-  'extends': 'eslint:recommended',
+  'extends': ['eslint:recommended',
+              'plugin:@typescript-eslint/recommended'
+  ],
+  'parser': '@typescript-eslint/parser',
+  'plugins': ['@typescript-eslint'],
   'parserOptions': {
     'sourceType': 'module',
-      "ecmaVersion": 8
+    'ecmaVersion': 8
   },
   'rules': {
+    'no-console': ['error'],
+    'no-var': ['error'],
+    'no-empty-function': ['error'],
+    'comma-dangle': ['error', 'never'],
+    'prefer-const': ['error'],
+    'quotes': ['error', 'single'],
+    'comma-spacing': ['error', { before: false, after: true }],
+    'semi-spacing': ['warn', { before: false, after: true }],
+    'space-before-blocks': ['warn', 'always'],
+    'switch-colon-spacing': ['warn', { after: true, before: false }],
+    'keyword-spacing': ['warn', { before: true, after: true }],
+    'template-curly-spacing': ['error', 'never'],
+    'rest-spread-spacing': ['error', 'never'],
+    'no-multi-spaces': ['warn', { ignoreEOLComments: false }],
```

Comment:
There are applied for consistent code style. The most prominent one is "prefer-const" to avoid any other `var` on the codebase.

Reply:
We could use prettier and use the default configuration to enforce it.  And configure husky